content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,213 @@
+1. [Configure OpenTelemetry](#step-3-configure-opentelemetry).
+
+### Step 1. Install up-to-date Prisma dependencies
+
+Use version `4.1.0` or higher of the `prisma`, `@prisma/client`, and '@prisma/instrumentation' npm packages.
+
+```terminal
+npm install prisma@latest --save-dev
+npm install @prisma/client@latest --save
+npm install @prisma/instrumentation --save
+```
+
+### Step 2: Enable the feature flag in your Prisma schema file
+
+In the `generator` block of your `schema.prisma` file, enable the `tracing` feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["tracing"]
+}
+```
+
+### Step 3: Configure OpenTelemetry
+
+First, install the appropriate OpenTelemetry packages, as follows:
+
+```console
+
+$ npm install @opentelemetry/api @opentelemetry/context-async-hooks @opentelemetry/exporter-trace-otlp-http @opentelemetry/instrumentation @opentelemetry/resources @opentelemetry/sdk-trace-base @opentelemetry/semantic-conventions
+
+```
+
+Then, add the following to your application code:
+
+```ts
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+
+export function otelSetup() {
+  registerInstrumentations({
+    instrumentations: [new PrismaInstrumentation()],
+  })
+}
+```
+
+## Send tracing to the console
+
+To send tracing to the console, add the following to you application code:
+
+```ts
+import { context } from '@opentelemetry/api'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+import {
+  BasicTracerProvider,
+  SimpleSpanProcessor,
+  ConsoleSpanExporter,
+} from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { PrismaClient } from '@prisma/client'
+
+const contextManager = new AsyncHooksContextManager().enable()
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Console exporter
+const consoleExporter = new ConsoleSpanExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+...
+```
+
+## Send tracing to Jaeger
```

Comment:
May want to include a docker command so they can run Jaeger locally.

Reply:
Docker command:
```
docker run --rm --name jaeger \
  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
  -e COLLECTOR_OTLP_ENABLED=true \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 4317:4317 \
  -p 4318:4318 \
  -p 14250:14250 \
  -p 14268:14268 \
  -p 14269:14269 \
  -p 9411:9411 \
  jaegertracing/all-in-one:1.35
```

Reply:
Thanks, added!